### PR TITLE
Pseudo-cherry-pick https://github.com/apple/swift/commit/96691208e9b8…

### PR DESCRIPTION
**Pull request by Catfish-Man**
Virtually all of the substantive changes made it over already, but a few were missed. This is ABI impacting, so we should take it for 5.0.

cc @airspeedswift

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Review by milseman — APPROVED** [review]
LGTM. Is this the entirety of the diff for these files?

**Comment by Catfish-Man** [comment]
Yup, that's all!

**Comment by airspeedswift** [comment]
@swift-ci please test

File: stdlib/public/core/StringBridge.swift [hunk 1]
```diff
@@ -302,12 +302,9 @@ extension String {
 // The @_swift_native_objc_runtime_base attribute
 // This allows us to subclass an Objective-C class and use the fast Swift
 // memory allocator.
-@_fixed_layout // FIXME(sil-serialize-all)
 @objc @_swift_native_objc_runtime_base(__SwiftNativeNSStringBase)
-public class __SwiftNativeNSString {
-  @usableFromInline // FIXME(sil-serialize-all)
-  @objc
-  internal init() {}
+class __SwiftNativeNSString {
```
Comment (airspeedswift) [inline stdlib/public/core/StringBridge.swift]: We should try and keep all access explicit, so this should have an `internal` (doesn't need to happen in this PR tho)

**Review by airspeedswift — APPROVED** [review]

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Pull request merged**